vdr/ambassador.go

```diff
@@ -221,6 +231,26 @@ func (n *ambassador) handleUpdateDIDDocument(transaction dag.SubscriberTransacti
 	return err
 }
 
+// missingTransactions does: current - incoming. Non conflicted updates will have an empty slice
+func missingTransactions(current []hash.SHA256Hash, incoming []hash.SHA256Hash) []hash.SHA256Hash {
+	j := 0
+	for _, h := range current {
+		found := false
+		for _, h2 := range incoming {
+			if h.Equals(h2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			current[j] = h
```

Comment:
slices are always passed by reference, so this mutates the `current` slice. Intended (potentially dangerous) side-effect?

Now I think of it, might a case where a deep copy from the in-memory DID store protects data corruption, if applicable?

Reply:
document and metadata are already a deepCopy when returned from the DB